### ISSUE TITLE
Issue with detecting high aspect ratio and small objects.

### ISSUE THREAD

**Issue by KaranBhuva22** [meituan/YOLOv6#1016]
### Before Asking

- [X] I have read the [README](https://github.com/meituan/YOLOv6/blob/main/README.md) carefully. 我已经仔细阅读了README上的操作指引。

- [X] I want to train my custom dataset, and I have read the [tutorials for training your custom data](https://github.com/meituan/YOLOv6/blob/main/docs/Train_custom_data.md) carefully and organize my dataset correctly; (FYI: We recommand you to apply the config files of xx_finetune.py.) 我想训练自定义数据集，我已经仔细阅读了训练自定义数据的教程，以及按照正确的目录结构存放数据集。（FYI: 我们推荐使用xx_finetune.py等配置文件训练自定义数据集。）

- [X] I have pulled the latest code of main branch to run again and the problem still existed. 我已经拉取了主分支上最新的代码，重新运行之后，问题仍不能解决。

### Search before asking

- [X] I have searched the YOLOv6 [issues](https://github.com/meituan/YOLOv6/issues) and found no similar questions.

### Question

I have fine-tuned YOLOv6 version 3 on custom datasets that include tables to identify headers, columns, rows, etc. However, during testing on validation/test data, I observed that the model struggles to detect objects with high aspect ratios, particularly the rows of tables.

To address this issue, I adjusted the area threshold within the box_candidate function in the data_augment.py file, which led to improvements in the results. Nevertheless, I am still encountering difficulties with wider objects.

![pt_WhatsApp Image 2023-02-23 at 12 57 47 PM_generated111_1](https://github.com/meituan/YOLOv6/assets/96902983/7952f261-2ffe-4491-9351-e33d1dd3ef9a)

![pt_Transaction_2380170281 - April 2022_generated91_1](https://github.com/meituan/YOLOv6/assets/96902983/4545b9f0-a5cb-4c1c-adae-24868e69306e)

![pt_Ravi Patel_IC-0412_pdf_2021-11-29_18_01_31_388241_jpg_2022-02-11_15_22_30_151398](https://github.com/meituan/YOLOv6/assets/96902983/cbffc6cd-8992-47b9-adbb-c2d47aac89f6)

![pt_01 To The New Private Limited - Apr-22 _generated53_1](https://github.com/meituan/YOLOv6/assets/96902983/5a746138-58c5-4ada-b61e-75fdfe077db3)

![pt_5 (18)_626227197189187_1](https://github.com/meituan/YOLOv6/assets/96902983/b722c6c1-3aa3-473a-86a1-20e598f3ca16)

The observed pattern is that the model accurately predicts some rows within the table but fails to identify several intermediate rows. For instance, if there are 10 rows in the table, the model may successfully predict 4 or 5 rows while overlooking the remaining 5 or 6 rows.

What are the possible ways for improvements ?

**Comment by Chilicyy** [comment]
Hi @KaranBhuva22 , it sounds like an interesting project. As for the performance of trained model, I wonder if it is related to the data and label distribution of training set.

**Comment by KaranBhuva22** [comment]
Hi @Chilicyy, 
I don't think it is related to data distribution issues because there was enough data for each class to fine-tune the model.

**Comment by Chilicyy** [comment]
Hi @KaranBhuva22 , did you doing training with fuse_ab mode open？

**Comment by KaranBhuva22** [comment]
Yes @Chilicyy fuse_ab model is open? I didn't modify it.

**Comment by Chilicyy** [comment]
@KaranBhuva22 Well, as for the problems you encountered, there's a few possible ways for improvements, I hope it helps.
1. With fuse_ab mode open, check anchors through AutoAnchor referring to [the feature from yolov5](https://github.com/ultralytics/yolov5/blob/875d92782e06dd16e2183b410ed9db18adf92e2e/train.py#L289C17-L289C30), and then modify the initialization of anchors in config file. 
2. Try to train from scratch instead of fine-tuning models, for the data distribution is rather different from COCO dataset.
3. Apply a larger scale of YOLOv6 model, such as yolov6n upgrade to yolov6s, to check if the model capacity is enough for your custom tasks.

**Comment by KaranBhuva22** [comment]
> @KaranBhuva22 Well, as for the problems you encountered, there's a few possible ways for improvements, I hope it helps.


Hello @Chilicyy, Thanks for the quick response.

> 1. With fuse_ab mode open, check anchors through AutoAnchor referring to [the feature from yolov5](https://github.com/ultralytics/yolov5/blob/875d92782e06dd16e2183b410ed9db18adf92e2e/train.py#L289C17-L289C30), and then modify the initialization of anchors in config file.
      
fuse_ab mode is not open since I am fine-tuning with `yolov6l6_finetune.py`, **Anchor Aided Training Mode** is currently unsupported on P6 models."


> 2. Try to train from scratch instead of fine-tuning models, for the data distribution is rather different from COCO dataset.

Which config file should I use for L6 model scratch training? Is there anything else I need to change except the configuration file for scratch training?

> 3. Apply a larger scale of YOLOv6 model, such as yolov6n upgrade to yolov6s, to check if the model capacity is enough for your custom tasks.

As I mentioned above, I am using the largest model, 'YOLOV6-L6', with 1280 resolution.
